src/psbt.cpp

```diff
@@ -531,12 +544,9 @@ bool PSBTOutput::Merge(const PSBTOutput& output)
 CTxOut PSBTOutput::GetTxOut() const
 {
```

Comment:
This would fail on situations where an asset was blinded but a value was not. I simplified the code to ensure that a value or commitment was checked in the assert (note: this will still crash if you don't provide either, maybe we should't assert?). Then it will use whatever combination of what is given.